lib/include/up-client-zenoh-cpp/transport/zenohUTransport.h

```diff
@@ -29,132 +29,126 @@
-                          z_encoding_t &encodingOut) noexcept;
-
-        /* zenoh session handle*/
-        z_owned_session_t session_;
-        /* indicate that termination is pending so no new transactions are allowed*/
-        atomic_bool termPending_ = false;
-        /* how many send transactions are currently in progress*/
-        atomic_uint32_t pendingSendRefCnt_ = 0;
-        /* how many times uTransport was initialized*/
-        atomic_uint32_t refCount_ = 0;
-        
-        std::mutex mutex_;
-
-        using uuriKey = size_t;     
-        using uuidStr = std::string;
-
-        unordered_map<uuriKey, z_owned_publisher_t> pubHandleMap_;
-        unordered_map<uuriKey, std::shared_ptr<ListenerContainer>> listenerMap_;  
-        unordered_map<uuriKey, bool> authorized_;
-        unordered_map<uuidStr, z_owned_query_t> queryMap_;
-
-        std::mutex pubInitMutex_;
-        std::mutex subInitMutex_;
-
-        static constexpr auto termMaxRetries_ = size_t(10);
-        static constexpr auto termRetryTimeout_ = std::chrono::milliseconds(100);
-
-        using cbArgumentType = std::tuple<std::shared_ptr<UUri>, ZenohUTransport*, const UListener&>;
-};
-
+namespace uprotocol::utransport {
+
+    class ListenerContainer {
+        public:
+            std::vector<z_owned_subscriber_t> subVector_;
+            std::vector<z_owned_queryable_t> queryVector_;
+            std::vector<const uprotocol::utransport::UListener*> listenerVector_;
+    };
+
+    class ZenohUTransport : public uprotocol::utransport::UTransport {
+
+        public:
+
+            ZenohUTransport(const ZenohUTransport&) = delete;
+            ZenohUTransport& operator=(const ZenohUTransport&) = delete;
+
+            /**
+            * The API provides an instance of the zenoh session
+            * @return instance of ZenohUTransport
+            */
+            static ZenohUTransport& instance(void) noexcept;
+
+            /**
+            * init the zenohUTransport 
+            * @return Returns OK on SUCCESS and ERROR on failure
+            */
+            uprotocol::v1::UStatus init() noexcept;
+
+            /**
+            * Terminates the zenoh utransport  - the API should be called by any class that called init
+            * @return Returns OK on SUCCESS and ERROR on failure
+            */
+            uprotocol::v1::UStatus term() noexcept; 
+
+            /**
+            * Transmit UPayload to the topic using the attributes defined in UTransportAttributes.
+            * @param topic Resolved UUri topic to send the payload to.
+            * @param payload Actual payload.
+            * @param attributes Additional transport attributes.
+            * @return Returns OKSTATUS if the payload has been successfully sent (ACK'ed), otherwise it
+            * returns FAILSTATUS with the appropriate failure.
+            */
+            uprotocol::v1::UStatus send(const uprotocol::v1::UUri &uri, 
```

Comment:
[Same comment about send() accepting a UMessage in the spec now](https://github.com/eclipse-uprotocol/up-client-zenoh-cpp/pull/21/files#r1528962948)

Reply:
David is working on this changes